pkg/expression/aggregation/aggregation.go

```diff
@@ -253,6 +256,22 @@ func CheckAggPushDown(ctx expression.EvalContext, aggFunc *AggFuncDesc, storeTyp
 	return ret
 }
 
+// checkVectorAggPushDown returns error if this aggregate function is not supported to push down.
+// - The aggregate function is not calculated over a Vector column (returns nil)
+// - The aggregate function is calculated over a Vector column and the function is supported (returns nil)
+// - The aggregate function is calculated over a Vector column and the function is not supported (returns error)
+func checkVectorAggPushDown(ctx expression.EvalContext, aggFunc *AggFuncDesc) error {
+	switch aggFunc.Name {
+	case ast.AggFuncCount, ast.AggFuncMin, ast.AggFuncMax, ast.AggFuncFirstRow:
+		return nil
+	default:
+		if aggFunc.Args[0].GetType(ctx).GetType() == mysql.TypeTiDBVectorFloat32 {
+			return errors.Errorf("Aggregate function %s is not supported for VectorFloat32", aggFunc.Name)
```

Comment:
Why return error instead of bool?

Reply:
/cc @breezewish  i think here can be changed into return bool？

Reply:
lgtm